### PR TITLE
Add IP Days Chip in Patient List and Patient Dashboard

### LINKED ISSUE #6684: Day of Admission: Patient
**Is your feature request related to a problem? Please describe.**
It is very useful for the users to understand the "Day of admission" for any patient. 

**Describe the solution you'd like**
Calculate the number of days from the day of admission and show the same on

- [ ] Patient dashboard
- [ ] Patient list

### PR DESCRIPTION
**Pull request by konavivekramakrishna**
## Proposed Changes

- Fixes #6684 
- Added IP days chip in patient list and patient dashboard
image
image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

**Review by rithviknishad — APPROVED** [review]

File: src/Utils/utils.ts [hunk 1]
```diff
@@ -479,3 +479,10 @@ export const mergeQueryOptions = <T extends object>(
     ),
   ];
 };
+
+export function daysUntilToday(inputDate: Date | string) {
+  return Math.floor(
+    (new Date().getTime() - new Date(inputDate).getTime()) /
+      (1000 * 60 * 60 * 24)
+  );
+}
```
Comment (Ashesh3) [inline src/Utils/utils.ts]:
```suggestion
export function daysUntilToday(inputDate: Date | string) {
  const date = dayjs(inputDate);
  return dayjs().diff(date, 'day');
}
```

Let's use dayjs for cleaner code.

**Review by Ashesh3 — CHANGES_REQUESTED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Comment by nihal467** [comment]
@konavivekramakrishna 
image

- [ ] Make the IP Days a circle badge on the consultation page similar to the MEWS score.

**Comment by vivramkrishna** [comment]
image


edge case when the year is 0202 it returns NaN

should this be handled?

**Comment by rithviknishad** [comment]
> edge case when the year is 0202 it returns NaN

can be ignored. newer data wouldn't have weird years right.

File: src/Components/Patient/PatientInfoCard.tsx [hunk 2]
```diff
@@ -465,6 +469,23 @@ export default function PatientInfoCard(props: {
...
           )}
+
+          <div>
+            <div className="  flex flex-col items-center   ">
+              <div className="border-grey-400 flex h-7 w-7 items-center justify-center rounded-full border-2 p-4 ">
+                <span className="text-sm font-semibold">
+                  {patient.last_consultation &&
+                  patient.last_consultation?.suggestion === "A"
+                    ? daysUntilToday(patient.last_consultation?.encounter_date)
+                    : "0"}
```
Comment (rithviknishad) [inline src/Components/Patient/PatientInfoCard.tsx]: Can we hide IP days for Non Admission Cases instead of showing 0?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
@konavivekramakrishna resolve the merge conflict

**Comment by konavivekramakrishna** [comment]
- image
- image

**Comment by nihal467** [comment]
@konavivekramakrishna 
image

the badge is missing text in the patient list view

**Comment by konavivekramakrishna** [comment]
I did't understand,

can you please specify what text it is missing,

IG everything is working as intended

**Comment by rithviknishad** [comment]
@nihal467 could you elaborate a bit more? Everything seems fine to me, IP Days are present in the patient list as well as consultation dashboard.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@rithviknishad can you push the two minor fix :

image

- [ ] add a prefix in the patient list page "IP Days"
- [ ] only show this badge if the patient is having a live consultation
- [ ] once the patient is discharged, in the consultation page retain number of days badge (not working now)

**Comment by rithviknishad** [comment]
Closing as completed in #7735

**Pull request closed without merging**